R/serializer.R

```diff
@@ -75,3 +75,29 @@ YAML <- R6::R6Class(
     }
   )
 )
+
+Compressed <- R6::R6Class(
+  "Compressed",
+  inherit = Serializer,
+  public = list(
+    initialize = function(path, name, preserve_bytes = FALSE) {
+      check_installed("qs2")
+      super$initialize(path, name, ".qs2", preserve_bytes = preserve_bytes)
+    },
+
+    serialize = function(data) {
+      out <- encode_interactions(data, self$preserve_bytes)
+      json <- as.character(jsonlite::toJSON(out, auto_unbox = TRUE))
```

Comment:
IMO it'd be better to save the R object directly. That will require changes to `decode_interactions()` to apply `sensitive_put_back()` in the opposite way that `encode_interactions()` does with `sensitive_remove()`. (That's a refactoring on my to do list)

Reply:
Okay, I did consider that, but yeah the `sensitive_put_back` issue. 

> (That's a refactoring on my to do list)

I'll leave a FIXME comment for now until that's addressed